lib/model.js

```diff
@@ -1404,6 +1410,16 @@ Model.prototype.findAll = function(options) {
       return this.runHooks('afterFind', results, options);
     }
   }).then(function (results) {
+
+    //kenophobic mode
+    if(_.isEmpty(results) && options.kenophobic) {
+      if( (typeof options.kenophobic === 'function')) {
+        throw new options.kenophobic();
```

Comment:
Not sure if I have done it in right way, but this will work for both cases now. Please check